ark/phenotyping/som_utils.py

```diff
@@ -238,6 +238,66 @@ def smooth_channels(fovs, tiff_dir, img_sub_folder, channels, smooth_vals):
                    chan_out, check_contrast=False)
 
 
+def filter_with_nuclear_mask(fovs, tiff_dir, seg_dir, channel=None,
+                             img_sub_folder=None, exclude=True):
+    """Filters out background staining using subcellular marker localization.
+
+    Non-nuclear signal is removed from nuclear markers and vice-versa for membrane markers.
+
+    Args:
+        fovs (list):
+            The list of fovs to filter
+        tiff_dir (str):
+            Name of the directory containing the tiff files
+        seg_dir (str):
+            Name of the directory containing the segmented files
+        channel (str):
+            Channel to apply filtering to
+        img_sub_folder (str):
+            Name of the subdirectory inside `tiff_dir` containing the tiff files.
+            Set to `None` if there isn't any.
+        exclude (bool):
+            Whether to filter out nuclear or membrane signal
+    """
+
+    # if seg_dir is None, the user cannot run filtering
+    if seg_dir is None:
```

Comment:
Since we don't ever set this to None, I don't see why the user would. Probably just checking that it's a valid directory makes more sense

Reply:
Users over on Marte's side have asked for `exampel_pixel_clustering.ipynb` support without a segmentation directory (ex. `seg_dir = None`), which Candace agreed to add in and provide support for in `create_pixel_matrix`. It might be worth discussing whether to continue supporting `seg_dir = None`, especially if we plan on adding more features requiring one.

Agreed that we do need to verify if `seg_dir` is a valid path.

Reply:
Ah you're right, we can keep it for now. There's definitely a use case for segmentation-free pixel clustering, we can give a more informative error message rather than failing with None as an invalid path